src/org/thoughtcrime/securesms/BaseActionBarActivity.java

```diff
@@ -38,26 +38,16 @@ protected void onCreate(Bundle savedInstanceState) {
       forceOverflowMenu();
     }
     super.onCreate(savedInstanceState);
-  }
-
-    @Override
-    protected void onStart() {
-        if (ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp() && !isWaitingForResult) {
-          ScreenLockUtil.applyScreenLock(this);
-        } else if (isHiddenByScreenLock) {
-          findViewById(android.R.id.content).setVisibility(View.VISIBLE);
-          isHiddenByScreenLock = false;
-        }
-        super.onStart();
-    }
 
-  @Override
-  protected void onStop() {
-    if (ScreenLockUtil.isScreenLockEnabled(this)) {
-      findViewById(android.R.id.content).setVisibility(View.GONE);
+    if (shouldLock()) {
+      // Already hide everything here to prevent sensible data from popping up shortly:
+      findViewById(android.R.id.content).setVisibility(View.INVISIBLE);
       isHiddenByScreenLock = true;
     }
-    super.onStop();
+  }
+
+  private boolean shouldLock() {
+    return ScreenLockUtil.isScreenLockEnabled(this) && ScreenLockUtil.getShouldLockApp() && !isWaitingForResult;
   }
 
   @Override
```

Comment:
I would move `isWaitingForResult = false;` in `onActivityResult` inside the `if (resultCode == RESULT_OK) {` block. But I didn't actually measure any difference when doing so. Maybe this is just the race-condition paranoia of mine.

Reply:
I should note that I am aware that you did not modify `onActivityResult`.